packages/block-editor/src/components/block-toolbar/index.js

```diff
@@ -33,9 +33,9 @@ function BlockToolbar( { blockClientIds, isValid, mode } ) {
 					<BlockSwitcher clientIds={ blockClientIds } />
 					<BlockControls.Slot />
 					<BlockFormatControls.Slot />
+					<BlockSettingsMenu clientIds={ blockClientIds } />
```

Comment:
This change seems to have a couple of unwanted side-effects:
- After selecting 'Edit as HTML' from the block settings menu, users are unable to switch back using the 'Edit visually' option because of the `mode === 'visual'` logic a few lines above.
- Users are unable to remove blocks that become invalid because of a block validation error (due to the `isValid` bool). This one's a little tricky to reproduce, I did the following:
1. Insert a seperator block and save the post
2. Modify the separator's save function so that it returns a `div` instead of an `hr`
3. Reload the post.
4. Observe the block validation error and the settings menu is missing.

Reply:
I see, thanks for spotting. It might actually be unrelated to the actual bug, so I‘ll try to revert this change

Reply:
Reverted in f5dce474e6e5b79ae299336d9ecca5382f60fb28